src/pages/SchedulePage.js

```diff
@@ -1,119 +1,280 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
+import { dayToStr } from "../constants";
+import WishlistCourse from "../components/WishlistCourse";
+import Calendar from "../components/Calendar";
+import ScheduleCourse from "../components/ScheduleCourse";
+
 import styled from "styled-components";
+import SnackBar from "@material-ui/core/Snackbar";
+import { grey } from "@material-ui/core/colors";
+
+import * as wishlistActions from "../redux/modules/wishlist";
+import * as courseActions from "../redux/modules/courseSelect";
+
+function SchedulePage({
+  year,
+  semester,
+  wishlist,
+  wishlistCourse,
+  scheduled,
+  toggleCourseSelect,
+  clearSchedule,
+}) {
+  const [schedule, setSchedule] = useState({});
+  const [checkboxes, setCheckboxes] = useState(
+    JSON.parse(localStorage.getItem(`${year}-${semester}-checkbox-state`)) || {}
+  );
+  const [Toast, setToast] = useState({
+    open: false,
+    message: "",
+    horizontal: "center",
+    vertical: "top",
+  });
+  const { open, message, horizontal, vertical } = Toast;
+
+  useEffect(() => {
+    (async () => {
+      try {
+        if (wishlist.length === 0) {
+          setSchedule({});
+          return;
+        }
+        if (scheduled.length === 0) {
+          setSchedule({});
+          return;
+        }
+        const response = await fetch(
+          `https://schedge.a1liu.com/${year}/${semester}/generateSchedule?registrationNumbers=${scheduled
+            .map((course) => course.courseRegistrationNumber)
+            .join(",")}`
+        );
+        if (!response.ok) {
+          return;
+        }
+        const data = await response.json();
+        if (!data.valid) {
+          setToast({
+            open: true,
+            message: `${data.conflictA.sectionName} & ${data.conflictB.sectionName} conflicts with one another!`, //make message more meaningful
+            horizontal: "center",
+            vertical: "top",
+          });
+          //Remove both conflicted course
+          let newCheckboxes = { ...checkboxes };
+          newCheckboxes[data.conflictA.registrationNumber] = false;
+          newCheckboxes[data.conflictB.registrationNumber] = false;
+          setCheckboxes(newCheckboxes);
+          localStorage.setItem(
+            `${year}-${semester}-checkbox-state`,
+            JSON.stringify(newCheckboxes)
+          );
+          toggleCourseSelect({
+            year,
+            semester,
+            conflicts: [
+              { courseRegistrationNumber: data.conflictA.registrationNumber },
+              { courseRegistrationNumber: data.conflictB.registrationNumber },
+            ],
+          });
+        } else {
+          setSchedule(data);
+        }
+        //handle layout data from schedge
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, [year, semester, scheduled]);
```

Comment:
add dependencies for `checkboxes`, `toggleCourseSelect`, and `wishlist.length` here